### ISSUE TITLE
Window dimensions doesn't change for android split screen

### ISSUE THREAD

**Issue by bigcupcoffee** [facebook/react-native#30371]
## Description
After upgrading react-native from 0.62.2 to 0.63.3 I've met an issue that when I use split screen on my physical android device the dimensions for window doesn't change. It remains the same if I open the app in fullscreen, or split it with other app so it takes 1/2 or 2/3 or 1/3 of the screen space. I cannot tell exactly what react-native version got this broken, but I did test it both on 0.62.2 and 0.63.3 versions once again, so I am confident about this being an issue on **latest** react-native version

## React Native version:
System:
    OS: Windows 10 10.0.19041
    CPU: (4) x64 Intel(R) Core(TM) i5-2400 CPU @ 3.10GHz
    Memory: 2.51 GB / 9.98 GB
  Binaries:
    Node: 14.5.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.0 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 6.14.5 - C:\Program Files\nodejs\npm.CMD
    Watchman: Not Found
  SDKs:
    Android SDK:
      API Levels: 28, 29
      Build Tools: 28.0.3, 29.0.2, 29.0.3
      System Images: android-21 | Google APIs ARM EABI v7a, android-21 | Google APIs Intel x86 Atom_64, android-27 | Google APIs Intel x86 Atom, android-28 | Intel x86 Atom_64
      Android NDK: Not Found
    Windows SDK: Not Found
  IDEs:
    Android Studio: Not Found
    Visual Studio: Not Found
  Languages:
    Java: Not Found
    Python: Not Found
  npmPackages:
    @react-native-community/cli: Not Found
    react: 16.13.1 => 16.13.1
    react-native: 0.63.3 => 0.63.3
    react-native-windows: Not Found

## Steps To Reproduce
I've left the code example down here, copy it and paste in App component of newly initializated react-native project on 0.63.3 version and try to open the app in fullscreen and split-screen, see the window height being the same(vertical split-screen).

## Expected Results
Dimensions.get('window').height AND useWindowDimensions hook must return different results for app being resized by using split-screen, but it returns the same value on & on, **but it works as expected in 0.62.2**

## Snack, code example, screenshot, or link to a repository:
```
  console.log('manual get height:', Dimensions.get('window').height);
  const {height} = useWindowDimensions();
  console.log('useWindowDimensions hook height:', height);
```

**Comment by fabriziobertoglio1987** [comment]
**<details><summary>CLICK TO OPEN TESTS RESULTS - </summary>**
<p>

In both cases I can see the update

| test 2 | 
|:-------------------------:|
| <video src="https://user-images.githubusercontent.com/24992535/110508294-29baf200-8101-11eb-9db8-c5bbb746976e.mp4" width="700" height="" /> |

| test 1 | 
|:-------------------------:|
| <video src="https://user-images.githubusercontent.com/24992535/110508290-29225b80-8101-11eb-8f05-c0aebe34f913.mp4" width="700" height="" /> |

</p>
</details>

**Comment by yfunk** [comment]
@fabriziobertoglio1987 Which version of React Native are you using in your demo?

@bigcupcoffee I've also been having this issue for a while on all different devices and emulators. I just stopped using the Dimensions API altogether because of this and because it also doesn't play nice with transparent navigation bars.

Instead  I've been using the [`useSafeAreaFrame()`](https://github.com/th3rdwave/react-native-safe-area-context#usesafeareaframe) hook from `react-native-safe-area-context`, which should already be included in a lot of projects (dependency of `react-navigation`). It works really well as a replacement if you make sure to only have one `SafeAreaProvider` at the top level of your app (or add a consumer at the top level and create your own hook).

**Comment by bigcupcoffee** [comment]
@yfunk yeah this is an another solution that's kinda working but there's also usecases for dimensions API for me. Moreover the API provided by react-native should be working fine so I'm still waiting for a fix and some of my apps still stay at 0.62 😬

**Comment by bigcupcoffee** [comment]
Bump: issue is still present in the latest release version(0.64.0)

**Comment by avertin** [comment]
This issue is also impacting my team. Really hoping it can be fixed in a future version of React Native

**Comment by bigcupcoffee** [comment]
Bump: persists on 0.68.0
Additionally noting that console.logs from example are generally stopping when changing split-screen height, like nothing re-renders(although scrollview gets proper height and scrolls to the end regardless). I'm definitely lacking some knowledge about changes in latest releases, would love if someone corrected me

**Comment by bigcupcoffee** [comment]
Bump: persists on 0.71.0
